azkaban-web-server/src/test/java/azkaban/imagemgmt/services/ImageRampRuleServiceImplTest.java

```diff
@@ -195,5 +200,40 @@ public void testNotAuthorizedUser() {
         .hasMessageContaining("unauthorized user");
   }
 
+  @Test
+  public void testAddFlowsToRule() {
+    final String json = JSONUtils.readJsonFileAsString("image_management/add_flows_to_rule.json");
+    final RampRuleFlowsDTO requestDTO = _converterUtils.convertToDTO(json, RampRuleFlowsDTO.class);
+    User user = new User("testAdminUser");
+    user.addRole("admin");
+    Role role = new Role(user.getUserId(), new Permission(Permission.Type.ADMIN));
+    when(_userManager.getRole(any())).thenReturn(role);
+
+    String existingOwner1 = "testAdminUser";
+    Set<String> existingOwners = new HashSet<>();
+    existingOwners.add(existingOwner1);
+    when(_rampRuleDao.getOwners(requestDTO.getRuleName())).thenReturn(existingOwners);
+    when(_projectLoader.isFlowInProject(any(), any())).thenReturn(true);
+    assertThatCode(() -> _rampRuleService.addFlowsToRule(requestDTO.getFlowIds(), requestDTO.getRuleName(), user))
+        .doesNotThrowAnyException();
+  }
+
+  @Test
+  public void testAddFlowsToRuleInvalidFlowId() {
```

Comment:
Please add a test for non-existent project.

Reply:
added